test/e2e/fishapp/dynamic_stack_test.go

```diff
@@ -0,0 +1,66 @@
+package fishapp_test
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/fishapp/shared"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/e2e/framework"
+	. "github.com/onsi/ginkgo"
+)
+
+var _ = Describe("tests by dynamic generated stack", func() {
+	var (
+		ctx          context.Context
+		f            *framework.Framework
+		dynamicStack *fishapp.DynamicStack
+	)
+
+	JustBeforeEach(func() {
+		ctx = context.Background()
+		f = framework.New(framework.GlobalOptions)
+		dynamicStack.Deploy(ctx, f)
+	})
+	JustAfterEach(func() {
+		dynamicStack.Cleanup(ctx, f)
+	})
+
+	Context("normal test dimensions with DNS", func() {
+		BeforeEach(func() {
+			dynamicStack = &fishapp.DynamicStack{
+				ServiceDiscoveryType:        shared.DNSServiceDiscovery,
+				VirtualServicesCount:        5,
+				VirtualNodesCount:           5,
+				RoutesCountPerVirtualRouter: 2,
+				TargetsCountPerRoute:        4,
+				BackendsCountPerVirtualNode: 2,
+				ReplicasPerVirtualNode:      3,
+				ConnectivityCheckPerURL:     100,
+			}
+		})
+		It("should behaves correctly", func() {
+			dynamicStack.ExpectBehavesCorrectly(ctx, f)
+		})
+	})
+
+	Context("normal test dimensions with cloudMap", func() {
+		BeforeEach(func() {
+			dynamicStack = &fishapp.DynamicStack{
+				ServiceDiscoveryType:        shared.CloudMapServiceDiscovery,
+				VirtualServicesCount:        5,
+				VirtualNodesCount:           5,
+				RoutesCountPerVirtualRouter: 2,
+				TargetsCountPerRoute:        4,
+				BackendsCountPerVirtualNode: 2,
+				ReplicasPerVirtualNode:      3,
+				ConnectivityCheckPerURL:     100,
+			}
+		})
+		It("should behaves correctly", func() {
+			dynamicStack.ExpectBehavesCorrectly(ctx, f)
+			var ans int
+			fmt.Println("press any thing to continue cleanup")
+			_, _ = fmt.Scan(&ans)
```

Comment:
Is there a reason to add this manual step here?

Reply:
nice catch, it's used for my debug steps to hold the test cleanup for trouble-shotting, removed :D